### ISSUE TITLE
Error: "OSError: [Errno Incorrect file size] deeplab_model/R-101-GN-WS.pth.tar" BrokenPipeError: [Errno 32] Broken pipe

### ISSUE THREAD

**Issue by eli5168** [royorel/FFHQ-Aging-Dataset#16]
```
Opening in existing browser session.
Authentication successful.
authorized access to google drive API!
Parsing JSON metadata...
Downloading 70001 files...
/ done processing 839/70001 filesTraceback (most recent call last):
  File "download_ffhq_aging.py", line 385, in <module>
    run_cmdline(sys.argv)
  File "download_ffhq_aging.py", line 380, in run_cmdline
    run(**vars(args))
  File "download_ffhq_aging.py", line 358, in run
    download_files(specs, dst_dir, output_size, check_invalid_images, drive=drive, **download_kwargs)
  File "download_ffhq_aging.py", line 219, in download_files
    raise exc_info[1].with_traceback(exc_info[2])
  File "download_ffhq_aging.py", line 229, in _download_thread
    pydrive_utils.pydrive_download(drive, spec['file_url'], spec['file_path'])
  File "/home/eli/Downloads/FFHQ-Aging-Dataset-master/pydrive_utils.py", line 40, in pydrive_download
    pydrive_file.GetContentFile(save_path)
  File "/home/eli/.local/lib/python3.6/site-packages/pydrive/files.py", line 210, in GetContentFile
    self.FetchContent(mimetype, remove_bom)
  File "/home/eli/.local/lib/python3.6/site-packages/pydrive/files.py", line 43, in _decorated
    return decoratee(self, *args, **kwargs)
  File "/home/eli/.local/lib/python3.6/site-packages/pydrive/files.py", line 255, in FetchContent
    self.content = io.BytesIO(self._DownloadFromUrl(download_url))
  File "/home/eli/.local/lib/python3.6/site-packages/pydrive/auth.py", line 75, in _decorated
    return decoratee(self, *args, **kwargs)
  File "/home/eli/.local/lib/python3.6/site-packages/pydrive/files.py", line 503, in _DownloadFromUrl
    resp, content = self.http.request(url)
  File "/home/eli/.local/lib/python3.6/site-packages/oauth2client/transport.py", line 175, in new_request
    redirections, connection_type)
  File "/home/eli/.local/lib/python3.6/site-packages/oauth2client/transport.py", line 282, in request
    connection_type=connection_type)
  File "/home/eli/.local/lib/python3.6/site-packages/httplib2/__init__.py", line 1702, in request
    conn, authority, uri, request_uri, method, body, headers, redirections, cachekey,
  File "/home/eli/.local/lib/python3.6/site-packages/httplib2/__init__.py", line 1421, in _request
    (response, content) = self._conn_request(conn, request_uri, method, body, headers)
  File "/home/eli/.local/lib/python3.6/site-packages/httplib2/__init__.py", line 1344, in _conn_request
    conn.request(method, request_uri, body, headers)
  File "/usr/lib/python3.6/http/client.py", line 1285, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.6/http/client.py", line 1331, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.6/http/client.py", line 1280, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.6/http/client.py", line 1046, in _send_output
    self.send(msg)
  File "/usr/lib/python3.6/http/client.py", line 1006, in send
    self.sock.sendall(data)
  File "/usr/lib/python3.6/ssl.py", line 975, in sendall
    v = self.send(byte_view[count:])
  File "/usr/lib/python3.6/ssl.py", line 944, in send
    return self._sslobj.write(data)
  File "/usr/lib/python3.6/ssl.py", line 642, in write
    return self._sslobj.write(data)
BrokenPipeError: [Errno 32] Broken pipe
Downloading backbone Resnet Model parameters
Traceback (most recent call last):
  File "run_deeplab.py", line 91, in <module>
    main()
  File "run_deeplab.py", line 50, in main
    download_file(session, resnet_file_spec)
  File "/home/eli/Downloads/FFHQ-Aging-Dataset-master/utils.py", line 70, in download_file
    raise IOError('Incorrect file size', file_path)
OSError: [Errno Incorrect file size] deeplab_model/R-101-GN-WS.pth.tar
```
There are some images generated successfully, but after a while something went wrong. So is this an issue with my network? Thanks.

**Comment by eli5168** [comment]
But I can still continue with the last download. So does this not affect the final result?

**Comment by eli5168** [comment]
```
Downloading 70001 files...
| done processing 1365/70001 filesTraceback (most recent call last):
  File "download_ffhq_aging.py", line 385, in <module>
    run_cmdline(sys.argv)
  File "download_ffhq_aging.py", line 380, in run_cmdline
    run(**vars(args))
  File "download_ffhq_aging.py", line 358, in run
    download_files(specs, dst_dir, output_size, check_invalid_images, drive=drive, **download_kwargs)
  File "download_ffhq_aging.py", line 219, in download_files
    raise exc_info[1].with_traceback(exc_info[2])
  File "download_ffhq_aging.py", line 229, in _download_thread
    pydrive_utils.pydrive_download(drive, spec['file_url'], spec['file_path'])
  File "/home/eli/Downloads/FFHQ-Aging-Dataset-master/pydrive_utils.py", line 40, in pydrive_download
    pydrive_file.GetContentFile(save_path)
  File "/home/eli/.local/lib/python3.6/site-packages/pydrive/files.py", line 210, in GetContentFile
    self.FetchContent(mimetype, remove_bom)
  File "/home/eli/.local/lib/python3.6/site-packages/pydrive/files.py", line 43, in _decorated
    return decoratee(self, *args, **kwargs)
  File "/home/eli/.local/lib/python3.6/site-packages/pydrive/files.py", line 255, in FetchContent
    self.content = io.BytesIO(self._DownloadFromUrl(download_url))
  File "/home/eli/.local/lib/python3.6/site-packages/pydrive/auth.py", line 75, in _decorated
    return decoratee(self, *args, **kwargs)
  File "/home/eli/.local/lib/python3.6/site-packages/pydrive/files.py", line 503, in _DownloadFromUrl
    resp, content = self.http.request(url)
  File "/home/eli/.local/lib/python3.6/site-packages/oauth2client/transport.py", line 175, in new_request
    redirections, connection_type)
  File "/home/eli/.local/lib/python3.6/site-packages/oauth2client/transport.py", line 282, in request
    connection_type=connection_type)
  File "/home/eli/.local/lib/python3.6/site-packages/httplib2/__init__.py", line 1702, in request
    conn, authority, uri, request_uri, method, body, headers, redirections, cachekey,
  File "/home/eli/.local/lib/python3.6/site-packages/httplib2/__init__.py", line 1421, in _request
    (response, content) = self._conn_request(conn, request_uri, method, body, headers)
  File "/home/eli/.local/lib/python3.6/site-packages/httplib2/__init__.py", line 1343, in _conn_request
    conn.connect()
  File "/home/eli/.local/lib/python3.6/site-packages/httplib2/__init__.py", line 1179, in connect
    raise socket_err
  File "/home/eli/.local/lib/python3.6/site-packages/httplib2/__init__.py", line 1135, in connect
    self.sock = self._context.wrap_socket(sock, server_hostname=self.host)
  File "/usr/lib/python3.6/ssl.py", line 407, in wrap_socket
    _context=self, _session=session)
  File "/usr/lib/python3.6/ssl.py", line 817, in __init__
    self.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 1077, in do_handshake
    self._sslobj.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 689, in do_handshake
    self._sslobj.do_handshake()
ConnectionResetError: [Errno 104] Connection reset by peer
Downloading backbone Resnet Model parameters
Traceback (most recent call last):
  File "run_deeplab.py", line 91, in <module>
    main()
  File "run_deeplab.py", line 50, in main
    download_file(session, resnet_file_spec)
  File "/home/eli/Downloads/FFHQ-Aging-Dataset-master/utils.py", line 70, in download_file
    raise IOError('Incorrect file size', file_path)
OSError: [Errno Incorrect file size] deeplab_model/R-101-GN-WS.pth.tar
```
There is a slight difference between this time and the last errors.

**Comment by eli5168** [comment]
```
Opening in existing browser session.
Authentication successful.
authorized access to google drive API!
Parsing JSON metadata...
Downloading 70001 files...
- done processing 3352/70001 filesTraceback (most recent call last):
  File "download_ffhq_aging.py", line 385, in <module>
  File "download_ffhq_aging.py", line 380, in run_cmdline
  File "download_ffhq_aging.py", line 358, in run
  File "download_ffhq_aging.py", line 219, in download_files
  File "download_ffhq_aging.py", line 234, in _download_thread
  File "download_ffhq_aging.py", line 330, in align_in_the_wild_image
    img.save(os.path.join(dst_subdir, '%05d.png' % item_idx))
  File "/usr/local/lib/python3.6/dist-packages/PIL/Image.py", line 2237, in save
    fp = builtins.open(filename, "w+b")
OSError: [Errno 24] Too many open files: 'ffhq_aging256x256/61000/61292.png'
Downloading backbone Resnet Model parameters
Traceback (most recent call last):
  File "run_deeplab.py", line 91, in <module>
    main()
  File "run_deeplab.py", line 50, in main
    download_file(session, resnet_file_spec)
  File "/home/eli/Downloads/FFHQ-Aging-Dataset-master/utils.py", line 70, in download_file
    raise IOError('Incorrect file size', file_path)
OSError: [Errno Incorrect file size] deeplab_model/R-101-GN-WS.pth.tar`
```

**Comment by royorel** [comment]
Hi @eli5168,

This issue is caused because the files were not downloaded successfully. It can stem from several issues.
1. Quota limit exceeded (a google drive issue, that cannot be bypassed)
2. Connectivity issues in you system

To check the first issue, you can go to Nvidia's google drive and try to manually download the image that caused the error. If you get a quota limit error, then that's the issue, otherwise, it's probably something local in your system.

**Comment by eli5168** [comment]
@royorel 

Thanks for your reply. I have successfully solved this problem.

Another question is if I only need to use the age and gender labels, does that mean I can just download and directly use 1024x1024 Aligned and Cropped images version FFHQ dataset (89.1 GB) with your FFHQ-Aging labels? Thanks.

**Comment by royorel** [comment]
@eli5168,

In our paper, we used different crop sizes than the original FFHQ dataset, which is why the code recreates the dataset from the in-the-wild images (the original images are deleted after the final image is created). The labels are still valid for the original 1024x1024, but since the crops are different, the results will be different from those of the paper.

**Comment by josejhlee** [comment]
@eli5168 How did you fix it? Just by manually downloading?
